cola/algorithms/slq.py

```diff
@@ -39,11 +39,12 @@ def slq_fwd(A, fun, num_samples, max_iters, tol, pbar, key):
     xnp = A.xnp
     rhs = xnp.randn(A.shape[1], num_samples, dtype=A.dtype, key=key, device=A.device)
     alpha, beta, _, iters, _ = lanczos_parts(A, rhs, max_iters, tol, pbar)
-    if xnp.__name__.find("torch") >= 0:
-        alpha, beta = alpha[..., :iters - 1], beta[..., :iters]
-    # REMOVED by marc due to it breaking jit compilation
-    else:
-        alpha = alpha[..., :-1]
+    # if xnp.__name__.find("torch") >= 0:
```

Comment:
@mfinzi , this logic of not trimming the coefficients was resulting in the log() function being evaluated in zeros for the test_logdet.

Reply:
that makes sense, but how do we fix it given that the function may need to be jittable?

Reply:
and even disregarding jit, won't we have the same problem if the batched lanczos procedure terminates after some alphas are 0 for some of the probe vectors?

Reply:
instead what if we add a
`log_eigvals = xnp.where(eigvals>1e-7*xnp.max(eigvals), fun(eigvals), xnp.zeros_like(eigvals)`
then those will not be included in the sum

Reply:
instead what if we add a
log_eigvals = xnp.where(eigvals>1e-6*xnp.max(eigvals), fun(eigvals), xnp.zeros_like(eigvals))
then those will not be included in the sum

And replace 1e-6 with 10*xnp.get_machine_eps(dtype)

Reply:
I changed this to your suggestion with a function that retrieves the machine precision values given the dtype